machine_learning_hep/processerdhadrons_jet.py

```diff
@@ -424,42 +424,41 @@ def process_efficiency_single(self, index):
                 bin_id = self.bin_matching[ipt]
                 if self.do_custom_analysis_cuts:
                     df_mc_reco = pickle.load(openfile(self.mptfiles_recosk[bin_id][index], "rb"))
-                    df_mc_reco["imp_par_prod"] = df_mc_reco["imp_par_prod"].astype(float) # allow negative cut values
+                    df_mc_reco["imp_par_prod"] = df_mc_reco["imp_par_prod"].astype(float)  # allow negative cut values
                 else:
                     df_mc_reco = pickle.load(openfile(self.mptfiles_recoskmldec[bin_id][index], "rb"))
                 df_mc_reco = adjust_nsd(df_mc_reco)
                 if self.s_evtsel is not None:
                     df_mc_reco = df_mc_reco.query(self.s_evtsel)
                 if self.s_jetsel_reco is not None:
-                    if self.p_usejetptbinned_deff is True:
-                        df_mc_reco = df_mc_reco.query(self.s_jetsel_reco)
+                    df_mc_reco = df_mc_reco.query(self.s_jetsel_reco)
                 if self.s_trigger is not None:
                     df_mc_reco = df_mc_reco.query(self.s_trigger)
                 if self.runlistrigger is not None:
                     df_mc_reco = selectdfrunlist(df_mc_reco, \
-                             self.run_param[self.runlistrigger], "run_number")
+                        self.run_param[self.runlistrigger], "run_number")
                 df_mc_gen = pickle.load(openfile(self.mptfiles_gensk[bin_id][index], "rb"))
                 df_mc_gen = adjust_nsd(df_mc_gen)
-                if self.p_usejetptbinned_deff is True:
+                if self.s_jetsel_gen is not None:
                     df_mc_gen = df_mc_gen.query(self.s_jetsel_gen)
-                df_mc_gen = df_mc_gen.query(cut_d0y)
+                df_mc_gen = df_mc_gen.query(self.s_jetsel_gen)
```

Comment:
`s_jetsel_gen` is applied twice here, once under condition, second time regardless.

Reply:
Fixed in https://github.com/nzardosh/MachineLearningHEP/pull/1